pkg/crane/copy.go

```diff
@@ -32,7 +32,7 @@ func Copy(src, dst string) error {
 		return fmt.Errorf("parsing reference %q: %v", src, err)
 	}
 
-	dstRef, err := name.ParseReference(dst)
+	dstRef, err := name.ParseWriteReference(dst)
```

Comment:
Since we discard the digest from the destination reference, we need to validate that the digest matches that of the image and return an error if it does not.